conformance/tests/httproute-invalid-cross-namespace-parent-ref.go

```diff
@@ -37,6 +39,17 @@ var HTTPRouteInvalidCrossNamespaceParentRef = suite.ConformanceTest{
 		gwNN := types.NamespacedName{Name: "same-namespace", Namespace: "gateway-conformance-infra"}
 		routeNN := types.NamespacedName{Name: "invalid-cross-namespace-parent-ref", Namespace: "gateway-conformance-web-backend"}
 
+		// The Route must have an Accepted Condition with a ParentRefNotPermitted Reason.
+		t.Run("HTTPRoute with a cross-namespace ParentRef where no ReferenceGrants allows such a reference, has an Accepted Condition with status False and Reason ParentRefNotPermitted", func(t *testing.T) {
```

Comment:
Sorry I'm late here -- aren't cross-namespace Route->Listener/Gateway attachments governed by the Listener's [AllowedRoutes](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.AllowedRoutes), not ReferenceGrant? The test name could use an update to clarify that, if I'm not mistaken.

Reply:
Honestly, I didn't realize that the grant was given by the allowedRoutes field, you are right. Other than changing the test name, the reason's documentation should also be updated, as it currently states https://github.com/kubernetes-sigs/gateway-api/blob/4ff6208a6a4238792e7477a03400933f5ae26c42/apis/v1beta1/shared_types.go#L252-L255

Reply:
Concerning this, are you aware if there is any specific reason not to use the `ReferenceGrants` in this case as well, in favor of `AllowedRoutes`?

Reply:
Actually, looking at this a bit more, could you clarify why the [NotAllowedByListeners](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.RouteConditionReason) reason was not appropriate here? It seems to me like it was intended for this scenario.

Reply:
ref. https://gateway-api.sigs.k8s.io/api-types/referencegrant/#exceptions

Reply:
> Actually, looking at this a bit more, could you clarify why the [NotAllowedByListeners](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.RouteConditionReason) reason was not appropriate here? It seems to me like it was intended for this scenario.

Yeah, this reason is not needed at all now that you point me out. The `NotAllowedByLysteners` reason should be used in this case.

Reply:
Thanks for catching this. I admittedly looked at this the first time like "yep!" and now I'm like "oh..." :fried_egg: . Thank you for all your contributions @mlavacca, stuff happens! And thank you @skriss for identifying it!

I'm reverting this and the other relevant PR, appreciate everyone's help :vulcan_salute:

Reply:
https://github.com/kubernetes-sigs/gateway-api/pull/1704 to follow up. Thank you!